src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -2660,6 +2660,55 @@ theorem msb_append {x : BitVec w} {y : BitVec v} :
   rw [getElem_append] -- Why does this not work with `simp [getElem_append]`?
   simp [show i < w by omega]
 
+theorem toNat_append_lt {m n : Nat} (x : BitVec m) (y : BitVec n) :
```

Comment:
```suggestion
/--
Helper lemma for `toInt_append` which establishes that
`(x.toNat <<< n) ||| `y.toNat` is within bounds of `BitVec (m + n)`.
-/
protected theorem toNat_shiftLeft_or_toNat_lt_two_pow_add {m n : Nat} (x : BitVec m) (y : BitVec n) :
```

Reply:
I renamed the theorem as suggested, added a comment, and also moved this theorem right before `toFin_append`. I did title this `helper lemma` as future users may be beyond `toFin_append`, but the closeness to `toFin_append` should now hopefully make this clear.